### ISSUE TITLE
Override XMLHttpRequest instead

### ISSUE THREAD

**Issue by gaearon** [travis-hilterbrand/chaos-monkey-browser#1]
This is a nice idea but the usability is very limited, as only jQuery AJAX calls are intercepted.
Instead, in my opinion, it should override `XMLHttpRequest`.

Any reasons for not doing this in the first place?

**Comment by travis-hilterbrand** [comment]
Yes!  Overriding XMLHTTPRequest is absolutely the best way to do this.  jquery still has a large install base so it's useful.  BUT, jquery isn't a necessity anymore.  I'm actively trying to remove it from all of my paying projects.

When I built this, I needed it right away and already had jquery-override code that made it simple.  

This seems like a good time to revisit this.  I'll try to find the time soon to rebuild without jquery.

Thanks for the suggestion!
